### PR TITLE
Support `Missing` in union attribute types

### PR DESCRIPTION
**Pull request by slwu89**
Mainly, adds support for attribute types which are union types including `Missing`. This entails adding specialized methods for `preimage` to deal with this case, as the usual `==` comparison doesn't work with the singleton object `missing`.

The cursed case where an attribute type is something like `Union{Nothing,Missing,...}` will still not work, but I think that is ok.

### CODE REVIEW

File: src/PreimageCaches.jl [hunk 1]
```diff
@@ -107,6 +107,14 @@ function preimage(dom, m::Mapping{S,T}, ::TrivialCache{S,T}, y) where {S,T}
   [x for x in dom if get(m, x, nothing) == y]
 end
 
+function preimage(dom, m::Mapping{S,T}, ::TrivialCache{S,T}, y::Missing) where {S,T>:Union{Missing}}
+  [x for x in dom if ismissing(get(m, x, nothing))]
+end
+
+function preimage(dom, m::Mapping{S,T}, ::TrivialCache{S,T}, y) where {S,T>:Union{Missing}}
+  [x for x in dom if ismissing(get(m, x, nothing)) ? false : get(m, x, nothing) == y]
```
Comment (epatters) [inline src/PreimageCaches.jl:115]: I think the conditional can be simplified to

```julia
coalesce(get(m, x, nothing) == y, false)
```

**Review by epatters — CHANGES_REQUESTED** [review]
Thanks Sean! One small comment to simplify the logic.

File: src/PreimageCaches.jl (hunk 1 above)
Comment (slwu89) [reply src/PreimageCaches.jl]: done! https://github.com/AlgebraicJulia/ACSets.jl/pull/58/commits/07fb97d5b3d24f6f1b0da0bdfca7ca11f9381a69

Very slick, I haven't seen `coalesce` before.

**Review by epatters — APPROVED** [review]
Thanks Sean, looks great!

**Pull request merged**